state/validation.go

```diff
@@ -177,23 +177,48 @@ func VerifyEvidence(stateDB dbm.DB, state State, evidence types.Evidence) error
 		)
 	}
 
+	if ev, ok := evidence.(*types.LunaticValidatorEvidence); ok {
+		if err := ev.VerifyHeader(committedHeader); err != nil {
+			return err
+		}
+	}
+
 	valset, err := LoadValidators(stateDB, evidence.Height())
 	if err != nil {
 		// TODO: if err is just that we cant find it cuz we pruned, ignore.
 		// TODO: if its actually bad evidence, punish peer
 		return err
 	}
 
-	// The address must have been an active validator at the height.
-	// NOTE: we will ignore evidence from H if the key was not a validator
-	// at H, even if it is a validator at some nearby H'
-	// XXX: this makes lite-client bisection as is unsafe
-	// See https://github.com/tendermint/tendermint/issues/3244
-	ev := evidence
-	height, addr := ev.Height(), ev.Address()
-	_, val := valset.GetByAddress(addr)
-	if val == nil {
-		return fmt.Errorf("address %X was not a validator at height %d", addr, height)
+	addr := evidence.Address()
+	var val *types.Validator
+
+	// For PhantomValidatorEvidence, check evidence.Address was not part of the
+	// validator set at height evidence.Height, but was a validator before OR
+	// after.
+	if phve, ok := evidence.(*types.PhantomValidatorEvidence); ok {
+		_, val = valset.GetByAddress(addr)
+		if val != nil {
+			return fmt.Errorf("address %X was a validator at height %d", addr, evidence.Height())
+		}
+
+		valset, err := LoadValidators(stateDB, phve.LastHeightValidatorWasInSet)
```

Comment:
check if it's still bonded (`LastHeightValidatorWasInSet < Height - MaxAgeNumInBlocks`)